src/Internal/DependencyManagement/CoreServiceProvider.php

```diff
@@ -139,6 +144,7 @@ class CoreServiceProvider extends AbstractServiceProvider {
 		SetupMerchantCenter::class    => true,
 		SetupCampaignNote::class      => true,
 		SetupCampaign2Note::class     => true,
+		SetupCouponSharing::class     => true,
```

Comment:
Preferable to use `SetupCouponSharingNote` (will require assignment alignment fix as well).
```suggestion
		SetupCouponSharingNote::class     => true,
```

Also, I've added a related note in `MerchantCenterAwareTrait` about injected `MerchantStatuses` directly.